bluebrain/repo-bluebrain/packages/coreneuron/package.py

```diff
@@ -40,7 +40,6 @@ class Coreneuron(CMakePackage):
     variant("tests", default=False, description="Enable building tests")
 
     # nmodl specific options
-    variant("nmodl", default=False, description="Use NMODL instead of MOD2C")
```

Comment:
Maybe we should delete the `CoreNEURON` recipe as well similarly to `mod2c` since it's not going to be used from now on?
Otherwise since the last version of `CoreNEURON` comes with `mod2c` as default I don't think it makes sense to remove the variant from here

Reply:
I initially thought to have it so that `mod2c` never existed to begin with and automatically use `nmodl`. Didn't get time to see it through

Reply:
Thinking about it, I'm leaning towards dropping coreneuron as well

Reply:
@pramodk do you see any issue with that?

Reply:
@iomaganaris: I am fine to remove coreneuron+mod2c recipe. Let me tag @olupton if he sees any near term needs of having these recipes around.

The only time we will need to go back to coreneuron+mod2c (/old spack recipes) is when we will move to RH v9 and then we have to redeploy (some of the) archive modules. It's not sure how we are going to do that though: use old spack version or use latest spack with old version of software packages. To be discussed and this would need a separate discussion.

Reply:
cc: @matz-e

Reply:
It seems a little odd to drop support for the latest release (8.2.2), but I suppose that in practice we have been exclusively deploying and using development pre-release versions on BB5 for some time, so for internal purposes this should be OK.

Reply:
I was also considering what would happen in systems without archives or if we merge stuff with upstream. Are we going to drop any versions of `NEURON`/`CoreNEURON` before `9.0.0`?

Reply:
> but I suppose that in practice we have been exclusively deploying and using development pre-release versions on BB5 for some time

thought the same!

Reply:
I think @iomaganaris's point is a good one; a recipe that only supports `neuron@9:` is not a credible contribution to upstream. So we can simplify our lives in the short term, but may be making more pain/excuses for @pramodk afterwards 🙃

Reply:
Upstream has no "CoreNEURON" package ^^

Reply:
the upstream recipe is a different/reduced version!